explorer/testdata/string/unicode.carbon

```diff
@@ -0,0 +1,25 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{explorer} %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes=false %s
+// RUN: %{explorer} --parser_debug --trace_file=- %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes %s
+// AUTOUPDATE: %{explorer} %s
+// CHECK: result: 0
+
+package ExplorerTest api;
+
+fn CompareStr(s: String) -> i32 {
+  if (s == "HELLO WORLD!!") {
+    return 0;
+  }
+  return 1;
+}
+
+fn Main() -> i32 {
+  Print("\u{0048}\u{0045}\u{004C}\u{004C}\u{004F} \u{0057}\u{004F}\u{0052}\u{004C}\u{0044}\u{0021}\u{21}\u{1F601}");
+  Print("\u{10FFFF}\u{0000}\u{00000000E9}\u{}");

```

Comment:
Note, printing null characters is going to mess with output. Please don't do this -- unit testing is better for checking things like this.

Reply:
I have shifted them to a new file, just to check if they are accepted or not.

Reply:
I think a better approach would be to write unit tests. If I understand correctly, this means writing more tests in the format shown in common/string_helpers_test.cpp. I've written a few such tests, and I will add them to your fork in a bit once I finish writing them @vinsdragonis

Reply:
I've finished my tests. @vinsdragonis , if you could add me as a collaborator to your fork, I can push them.

Reply:
Sent invite

Reply:
Looks like we're still having some errors, and I think those are because of the test fail_unicode.carbon. I think we can remove this test, since all of the features it tests are already in the unit tests I added.

Reply:
Figures why I had a problem with the passing of the checks. I'll remove it.

Reply:
@Pritjam you were right. The checks didn't pass only because of `fail_unicode.carbon`.